engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -994,33 +994,28 @@ public Answer easySend(final Long hostId, final Command cmd) {
     }
 
     @Override
-    public boolean reconnect(final long hostId) {
+    public void reconnect(final long hostId) throws CloudRuntimeException, AgentUnavailableException{
         HostVO host;
 
         host = _hostDao.findById(hostId);
         if (host == null || host.getRemoved() != null) {
-            s_logger.warn("Unable to find host " + hostId);
-            return false;
+            throw new CloudRuntimeException("Unable to find host " + hostId);
         }
 
         if (host.getStatus() == Status.Disconnected) {
-            s_logger.info("Host is already disconnected, no work to be done");
-            return true;
+            throw new CloudRuntimeException("Host is already disconnected, no work to be done");
```

Comment:
This looks wrong to me, why throwing an exception when nothing has to be done?

Reply:
I agree on this one. And besides that, if we take a closer look, the guy that proposed this change, was replacing the return of boolean values with exceptions to indicate some error. It makes sense that a false return would generate an exception while trying to reconnect as the case at line 1002. This one was returning true, so it feels that we should not throw an exception here.